test/distributed/tensor/test_dtensor.py

```diff
@@ -956,7 +956,7 @@ def test_split_tensor_1D(self) -> None:
             )
             if size == 0:
                 # when tensor size is 0, there is no padding needed for all the ranks.
-                expected_pad_sizes = []
+                expected_pad_sizes = [0] * self.world_size
```

Comment:
I'm changing this calling convention, as it makes for a simpler implementation of `_split_tensor` _and_ for a simpler callsite logic (no more checking whether pad_sizes is empty). Since `_split_tensor` is private I don't believe this is really a BC-change.